pages/docs/reference/compatibility.md

```diff
@@ -0,0 +1,114 @@
+
+- **C** - Full **C**ompatibility
+  - Language
+    - no syntax changes (*modulo bugs**)
+    - new warnings/hints may be added or removed
+  - API (`kotlin-stdlib-*`, `kotlin-reflect-*`)
+    - no API changes
+    - deprecations with level `WARN` may be added/removed
+  - Binaries (ABI)
+    - runtime: binaries can be used interchangeably
+    - compilation: binaries can be used interchangeably
+- **BCL** - **B**ackward **C**ompatibility for the **L**anguage
+  - Language
+    - syntax deprecated in OV may be removed in NV
+    - other than that, all code compilable in OV is compilable by in NV (modulo bugs*)
+    - new syntax may be added in NV
+    - some restrictions of OV may be lifted in NV 
+    - new warnings/hints may be added or removed
+  - API (`kotlin-stdlib-*`, `kotlin-reflect-*`)
+    - new APIs may be added
+    - deprecations with level `WARN` may be added/removed
+    - deprecations with level `WARN` may be elevated to level `ERROR` or `HIDDEN` in NV
+- **BCB** - Full **B**ackward **C**ompatibility for Binaries
+  - Binaries (ABI)
+    - runtime: NV-binaries can be used  everywhere where OV binaries worked
+    - NV compiler: code compilable against OV binaries is compilable against NV binaries
+    - OV compiler may not accept NV binaries (e.g. those that exhibit newer language features or APIs)
+- **BC** = BCL & BCB
+- **EXP** - Experimental feature
+  - see below
+- **NO** - No compatibility guarantees
+  - we'll do our best to offer smooth migration, but can give no guarantees
+  - migration is planned individually for every incompatible subsystem 
+
+---
+
+\* No changes *modulo bugs* means that if an important bug is found (e.g. in the compiler diagnostics or elsewhere), a fix for it may introduce a breaking change, but we are always very careful with such changes.
+
+## Compatibility guarantees for Kotlin releases
+
+**Kotlin for JVM**:
+  - patch version updates (e.g. 1.1.X) are fully compatible
+  - minor version updates (e.g. 1.X) are backwards compatible
+
+| JVM  | 1.0 | 1.0.X | 1.1 | 1.1.X | ... | 2.0 |
+|---:|:---:|:---:|:---:|:---:|:---:|:---:|
+| **1.0** | - | C | BC | BC | ... | ? |
+| **1.0.X** | C | - | BC | BC | ... | ?
+| **1.1** | BC | BC | - | C | ... | ?
+| **1.1.X** | BC | BC | C | - | .... | ?
+| **...** | ... | ... | ... | ... | ... | ... |
+| **2.0** | ? | ? | ? | ? | ... | - 
+
+**Kotlin for JS**: starting with Kotlin 1.1, both patch and minor version updates provide backward compatibility for the language (BCL), but no BCB.  
+
+| JS  | 1.0.X | 1.1 | 1.1.X | ... | 2.0 |
+|---:|:---:|:---:|:---:|:---:|:---:|
+| **1.0.X** | - |  EXP | EXP | ... | EXP |
```

Comment:
I don't think I understand the difference between EXP and NO here, especially because the table for experimental feature compatibility below uses NO, not EXP